tests/test_embeds.py

```diff
@@ -289,6 +289,79 @@ def test_fields_exceptions() -> None:
         embed.set_field_at(0, "x", "y")
 
 
+def test_field_restraints() -> None:
+    embed = Embed(title="T" * 256)  # Size = 256
+    embed.set_footer(text="T" * 2048)  # Size = 2304
+    embed.add_field(name="A" * 256, value="B" * 1024)  # Size = 3584
+    embed.add_field(name="A" * 256, value="B" * 1024)  # Size = 4864
+    embed.add_field(name="A" * 112, value="B" * 1024)  # Size = 6000
+    assert len(embed) == 6000
+    assert embed.check_limits()
+
+    embed.add_field(name="A", value="B")  # Breaks limit of 6000 chars
+    with pytest.raises(ValueError, match="^Embed"):
```

Comment:
These matches should probably be more specific, to check that it raises the correct length error.

Reply:
https://github.com/DisnakeDev/disnake/pull/567/commits/e7bd728976f670dd731f784a659d8ca1064832e4